atlasdb-container-test-utils/src/main/java/com/palantir/atlasdb/containers/ThreeNodeCassandraCluster.java

```diff
@@ -60,12 +80,121 @@ public String getDockerComposeFile() {
     }
 
     @Override
-    public SuccessOrFailure isReady() {
+    public SuccessOrFailure isReady(DockerComposeRule rule) {
         return SuccessOrFailure.onResultOf(() -> {
-            CassandraKeyValueService.create(
-                    CassandraKeyValueServiceConfigManager.createSimpleManager(KVS_CONFIG),
-                    LEADER_CONFIG);
-            return true;
+
+            try {
+                if (!nodetoolShowsThreeCassandraNodesUp(rule)) {
+                    return false;
+                }
+
+                // slightly hijacking the isReady function here - using it
+                // to actually modify the cluster
+                replicateSystemAuthenticationDataOnAllNodes(rule);
+
+                return canCreateCassandraKeyValueService();
+            } catch (Exception e) {
+                log.info("Exception while checking if the Cassandra cluster was ready: " + e);
+                return false;
+            }
         });
     }
+
+    private boolean canCreateCassandraKeyValueService() {
```

Comment:
Maybe it would be better to move them all out into a separate class that takes in `DockerComposeRule` and number of nodes.